subsys/bluetooth/controller/ll_sw/ull_conn.c

```diff
@@ -331,6 +337,12 @@ u8_t ll_feature_req_send(u16_t handle)
 		return BT_HCI_ERR_CMD_DISALLOWED;
 	}
 
+	conn->llcp_req++;
```

Comment:
On Armv7-M this would be better served by using `atomic_cas()` is my opinion. I am not sure about other architectures (ping @mped-oticon and @aescolar) but on Armv7-M `atomic_cas()` uses `CONFIG_ATOMIC_OPERATIONS_BUILTIN` which then translates it into the following, which in my opinion is going to be smaller in size than your code once assembled.
Note that on Armv6-M `atomic_cas()` disables interrupts and so there you will need the software implementation you have here.

```
        return __atomic_compare_exchange_n(target, &old_value, new_value,
     4d4:       4b17            ldr     r3, [pc, #92]   ; (534 <bt_ready+0x74>)
     4d6:       9403            str     r4, [sp, #12]
     4d8:       2104            movs    r1, #4
     4da:       f3bf 8f5b       dmb     ish
     4de:       e853 2f00       ldrex   r2, [r3]
     4e2:       42a2            cmp     r2, r4
     4e4:       d103            bne.n   4ee <bt_ready+0x2e>
     4e6:       e843 1000       strex   r0, r1, [r3]
     4ea:       2800            cmp     r0, #0
     4ec:       d1f7            bne.n   4de <bt_ready+0x1e>
```

Reply:
Note: Oticon's mcu does not have atomic_cas.

Reply:
Since the value is incremented, should we use `atomic_inc(&conn->llcp_req)` ?